tests/unit/src/test/scala/tests/DocumentSymbolSlowSuite.scala

```diff
@@ -0,0 +1,66 @@
+package tests
+
+object DocumentSymbolSlowSuite extends BaseSlowSuite("documentSymbol") {
+
+  testAsync("parse-error") {
+    for {
+      // start with code that does not parse (notice the first char in Main.scala)
+      _ <- server.initialize(
+        """|
+           |/metals.json
+           |{
+           |  "a": { }
+           |}
+           |/a/src/main/scala/a/Main.scala
+           |} // <- parse error
+           |object Outer {
+           |  class Inner
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/Main.scala")
+      // check that no document symbols have been found for the unparseable code
+      _ = assertNoDiff(
+        server.documentSymbols("a/src/main/scala/a/Main.scala"),
+        """|} // <- parse error
+           |object Outer {
+           |  class Inner
+           |}""".stripMargin
+      )
+      // fix the code to make it parse
+      _ <- server.didChange("a/src/main/scala/a/Main.scala") { text =>
+        text.replaceFirst("}", "")
```

Comment:
Also works:
```scala
        """|object Outer {
           |  class Inner
           |}""".stripMargin
```

Reply:
whichever you think is clearer, I have no preference

Reply:
how about both?

Example:

```scala
/*example(Package):24*/package example

/*PatternMatching(Class):24*/class PatternMatching {
  /*some(Constant):4*/val some = Some(1)
  some match {
    case Some(number) =>
      number
  }

  // tuple deconstruction
  /*left(Constant):11*//*right(Constant):11*/val (left, right) = (1, 2)
  (left, right)

  // val deconstruction
  /*number1(Constant):16*/val Some(number1) =
    some
  println(number1)

  /*localDeconstruction(Method):23*/def localDeconstruction = {
    val Some(number2) =
      some
    number2
  }
}
```

Reply:
👍